hub/core/dataset/dataset.py

```diff
@@ -1138,7 +1142,49 @@ def delete(self, large_ok=False):
         self._unlock()
         self.storage.clear()
 
+    def summary(self):
+        head = [
+            "tensor",
+            "htype",
+            "shape",
+            "dtype",
+            "compression",
+        ]  # To add more attributes change head, divider, row_array and table_str
+        divider = ["-------"] * 5
+        tensor_dict = self.version_state[
+            "full_tensors"
+        ]  # Creating a list of tensors in the dataset
+        maxColumnLength = [7, 7, 7, 7, 7]  # length of "attribute name" length
+        count = 0
+        tableArray = [head, divider]  # collects all the rows
+        for tensor in tensor_dict:
+            tensor_object = tensor_dict[tensor]
+
+            tensor_name = tensor
+            tensor_htype = tensor_object.htype
+            tensor_shape = str(tensor_object.shape)
+            tensor_dtype = tensor_object.dtype.name
```

Comment:
`tensor_object.dtype` could be None and it will not have `name` attribute. Just do `ds.create_tensor("abc")` to reproduce the error.

Reply:
I made the changes. Can you please review it once again? @FayazRahman @farizrahman4u @AbhinavTuli @mikayelh